### PR TITLE
Bug 573296 - [Passage] rework EMF migration facilities, part 10

### PR DESCRIPTION
**Pull request by ruspl-afed**
Rework "delegating" classes

Signed-off-by: Alexander Fedorov <alexander.fedorov@arsysop.ru>

### CODE REVIEW

File: bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/migration/ApplyFeatureMap.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.Optional;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.util.FeatureMap;
+import org.eclipse.emf.ecore.util.FeatureMap.Entry;
+
+/**
+ * @since 2.0
+ */
+public final class ApplyFeatureMap {
```
Comment (eparovyshnaya) [inline bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/migration/ApplyFeatureMap.java:26]: Consumer<EObject>?

File: bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/internal/emf/migration/DelegatingEPackage.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lic.internal.emf.migration;
+
+import org.eclipse.emf.ecore.impl.EPackageImpl;
+
+final class DelegatingEPackage extends EPackageImpl {
+
+	private final DelegatingEFactory delegatingEFactory = new DelegatingEFactory();
```
Comment (eparovyshnaya) [inline bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/internal/emf/migration/DelegatingEPackage.java:19]: just `factory`?

**Review by eparovyshnaya — APPROVED** [review]

File: bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/migration/ApplyFeatureMap.java (hunk 1 above)
Comment (ruspl-afed) [reply bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/migration/ApplyFeatureMap.java:26]: may be later

File: bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/internal/emf/migration/DelegatingEPackage.java (hunk 2 above)
Comment (ruspl-afed) [reply bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/internal/emf/migration/DelegatingEPackage.java]: done

**Pull request merged**